tableauserverclient/models/schedule_item.py

```diff
@@ -0,0 +1,225 @@
+        else:
+            self._execution_order = value
+
+    @property
+    def frequency(self):
+        return self._frequency
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        if not value:
+            error = "Name must be defined."
+            raise ValueError(error)
+        else:
+            self._name = value
+
+    @property
+    def next_run_at(self):
+        return self._next_run_at
+
+    @property
+    def priority(self):
+        return self._priority
+
+    @priority.setter
+    def priority(self, value):
+        if value < 1 or value > 100:
+            error = "Invalid priority defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._priority = value
+
+    @property
+    def schedule_type(self):
+        return self._schedule_type
+
+    @schedule_type.setter
+    def schedule_type(self, value):
+        if not value:
+            error = "Schedule type must be defined."
+            raise ValueError(error)
+        elif not hasattr(ScheduleItem.Type, value):
+            error = "Invalid schedule type defined: {}.".format(value)
+            raise ValueError(error)
+        else:
+            self._schedule_type = value
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        if not hasattr(ScheduleItem.State, value):
+            error = "Invalid state defined."
+            raise ValueError(error)
+        else:
+            self._state = value
+
+    @property
+    def updated_at(self):
+        return self._updated_at
+
+    def _parse_common_tags(self, schedule_xml):
+        if not isinstance(schedule_xml, ET.Element):
+            schedule_xml = ET.fromstring(schedule_xml).find('.//t:schedule', namespaces=NAMESPACE)
+        if schedule_xml is not None:
+            (_, name, _, _, updated_at, _, frequency, next_run_at, end_schedule_at, execution_order,
+             priority, interval_item) = self._parse_element(schedule_xml)
+
+            self._set_values(id=None,
+                             name=name,
+                             state=None,
+                             created_at=None,
+                             updated_at=updated_at,
+                             schedule_type=None,
+                             frequency=frequency,
+                             next_run_at=next_run_at,
+                             end_schedule_at=end_schedule_at,
+                             execution_order=execution_order,
+                             priority=priority,
+                             interval_item=interval_item)
+
+        return self
+
+    def _set_values(self, id, name, state, created_at, updated_at, schedule_type, frequency,
+                    next_run_at, end_schedule_at, execution_order, priority, interval_item):
+        if id is not None:
```

Comment:
I feel like accepting **kwargs and then having a list of accepted attributes and iterating over them might be cleaner than all of these.

```python
# Assume some static value called "ALLOWED_ATTRIBUTES"
def _set_values(self, **kwargs):
    for attr_name in self.ALLLOWED_ATTRIBUTES:
        if attr_name in kwargs and kwargs[attr_name]:
            setattr(self, '_{}'.format(attr_name), kwargs[attr_name]
```

Since this is mostly a helper method, I think it's okay to use the kwargs